centreon/src/Core/TimePeriod/Infrastructure/Repository/DbReadTimePeriodRepository.php

```diff
@@ -0,0 +1,316 @@
+            'id' => 'tp_id',
+            'name' => 'tp_name',
+            'alias' => 'tp_alias',
+        ]);
+        $request = $this->translateDbName('SELECT SQL_CALC_FOUND_ROWS tp.* FROM `centreon`.timeperiod tp');
+
+        // Search
+        $request .= $sqlRequestTranslator->translateSearchParameterToSql();
+
+        // Sort
+        $sortRequest = $sqlRequestTranslator->translateSortParameterToSql();
+        $request .= !is_null($sortRequest)
+            ? $sortRequest
+            : ' ORDER BY tp_id ASC';
+
+        // Pagination
+        $request .= $sqlRequestTranslator->translatePaginationToSql();
+
+        $statement = $this->db->prepare($request);
+        foreach ($sqlRequestTranslator->getSearchValues() as $key => $data) {
+            $type = key($data);
+            if ($type !== null) {
+                $value = $data[$type];
+                $statement->bindValue($key, $value, $type);
+            }
+        }
+        $statement->execute();
+        $result = $this->db->query('SELECT FOUND_ROWS()');
+        if ($result !== false && ($total = $result->fetchColumn()) !== false) {
+            $sqlRequestTranslator->getRequestParameters()->setTotal((int) $total);
+        }
+        /**
+         * @var array<int, TimePeriod> $timePeriods
+         */
+        $timePeriods = [];
+
+        while (($result = $statement->fetch(\PDO::FETCH_ASSOC)) !== false) {
+            /**
+             * @var array{
+             *     tp_id: int,
+             *     tp_name: string,
+             *     tp_alias: string,
+             *     tp_monday: string,
+             *     tp_tuesday: string,
+             *     tp_wednesday: string,
+             *     tp_thursday: string,
+             *     tp_friday: string,
+             *     tp_saturday: string,
+             *     tp_sunday: string,
+             *     template_id: int
+             * } $result
+             */
+            $timePeriod = $this->createTimePeriod($result);
+            $timePeriods[$result["tp_id"]] = $timePeriod;
+        }
+
+        $this->addTemplates($timePeriods);
+        $this->addExtraTimePeriods($timePeriods);
+
+        return $timePeriods;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function nameAlreadyExists(string $timePeriodName, int $timePeriodId = null): bool
+    {
+
+        $statement = $this->db->prepare(
+            $this->translateDbName('SELECT tp_id FROM `:db`.timeperiod WHERE tp_name = :name')
+        );
+        $statement->bindValue(':name', $timePeriodName);
+        $statement->execute();
+        /**
+         * @var array{tp_id: int}|false $result
+         */
+        $result = $statement->fetch(\PDO::FETCH_ASSOC);
+        if ($timePeriodId !== null) {
+            if ($result !== false) {
+                return $result['tp_id'] != $timePeriodId;
+            }
+            return false;
+        } else {
+            return ! (empty($result));
+        }
```

Comment:
For better comprehension, maybe inverse the conditions ? (early-return "pattern")
```suggestion
        if ($timePeriodId === null) {
            return ! (empty($result));
        }

        if ($result === false) {
            return false;
        }

        return $result['tp_id'] !== $timePeriodId;
```